File: BOJ/FastCampas_Algorithm_lec/BOJ_1874/스택수열.cpp
#include<iostream>
#include<stack>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int n, a, cnt = 1;
	string ans = "";
	stack<int> st;
	cin >> n;
	for (int i = 1; i <= n; i++) {
		cin >> a;
		while (cnt <= a) {
			st.push(cnt);
			cnt += 1;
			ans += "+";
			ans += "\n";
		}
		if (st.top() == a) {
			st.pop();
			ans += "-";
			ans += "\n";
		}
		else {
			cout << "NO" << "\n";
			return 0;
		}
	}
	cout << ans << "\n";
	return 0;
}

File: BOJ/FastCampas_Algorithm_lec/BOJ_1874/스택수열.py
n = int(input())

cnt = 1
stack = []
result = []

for i in range(1, n+1):
    num = int(input())
    while cnt <= num:
        stack.append(cnt)
        cnt += 1
        result.append('+')
    if stack[-1] == num:
        stack.pop()
        result.append('-')
    else:
        print('NO')
        exit(0)
print('\n'.join(result))
